src/content/components/FlameChartCanvas.js

```diff
@@ -173,7 +177,59 @@ class FlameChartCanvas extends PureComponent {
       durationString = duration.toFixed(3);
     }
 
-    return `${durationString}ms - ${label}`;
+    const stackIndex = stackTiming.stack[stackTableIndex];
+    const frameIndex = thread.stackTable.frame[stackIndex];
+    const label = getLabel(thread, stackIndex);
+    const category = getCategory(thread, frameIndex);
+    const funcIndex = thread.frameTable.func[frameIndex];
+
+    let resourceOrFileName = null;
+    // Only show resources or filenames if the chart is expanded, as collapsed stacks
+    // would show incorrect details about a group of stacks.
+    if (isRowExpanded) {
+      // Only JavaScript functions have a filename.
+      const fileNameIndex = thread.funcTable.fileName[funcIndex];
+      if (fileNameIndex !== null) {
+        resourceOrFileName = (
+          <div className='tooltipOneLine tooltipDetails'>
+            <div className='tooltipLabel'>File:</div>
+            {thread.stringTable.getString(fileNameIndex)}
+          </div>
+        );
+      } else {
+        const resourceIndex = thread.funcTable.resource[funcIndex];
+        if (resourceIndex !== -1) {
+          const resourceNameIndex = thread.resourceTable.name[resourceIndex];
+          if (resourceNameIndex !== -1) {
+            resourceOrFileName = (
+              <div className='tooltipOneLine tooltipDetails'>
+                <div className='tooltipLabel'>Resource:</div>
+                {thread.stringTable.getString(resourceNameIndex)}
+              </div>
+            );
+          }
+        }
+      }
+    }
+
+    return (
+      <div className='flameChartCanvasTooltip'>
+        <div className='tooltipOneLine tooltipHeader'>
+          <div className='tooltipTiming'>
+            {durationString}ms
+          </div>
+          <div className='tooltipName'>
+            {label}
+          </div>
+        </div>
+        <div className='tooltipOneLine tooltipDetails'>
+          <div className='tooltipLabel'>Category:</div>
+          <div className='tooltipSwatch' style={{backgroundColor: category.color}} />
+          {category.name}
+        </div>
+        {resourceOrFileName}
+      </div>
```

Comment:
Hmm... I'm not really sure how to handle that. I'd like the Tooltip to only handle the tooltip logic. I'd like the presentation to be free-form and up to the controlling element. When I'm implementing something per-panel I'd prefer to do what I want. Placing some shared style in the Tooltip css seemed the most convenient place.